networkx/classes/function.py

```diff
@@ -795,6 +795,34 @@ def set_edge_attributes(G, values, name=None):
         >>> (1, 2) in G.edges()
         False
 
+    For multigraphs, the `values` dict is expected to be keyed by 3-tuples
+    including the edge key::
+
+        >>> G = nx.MultiGraph()
+        >>> edges = [(0, 1), (0, 1)]
+        >>> G.add_edges_from(edges)  # Returns list of edge keys
+        [0, 1]
+        >>> attributes = {(0, 1, 0): {"cost": 21}, (0, 1, 1): {"cost": 7}}
+        >>> nx.set_edge_attributes(G, attributes)
+        >>> G[0][1][0]["cost"]
+        21
+        >>> G[0][1][1]["cost"]
+        7
+
+    The Attributes of one Graph can be set for another. It doesn't matter the
+    exact type of Graph. If its Multigraph attributes assigned to a Path Graph
+    with same nodes, the edge would take the last attribute value being passed
+    for the same edge. If the edge doesn't exist it is simpy ignored. An example
+    continuing from the previous case is shown below::
+
+        >>> H = nx.path_graph([0, 1, 2])
+        >>> attributesH = {(u, v) : attributes for u, v, attributes in list(G.edges(data = True))}
+        >>> attributesH
+        {(0, 1): {'cost': 7}}
+        >>> nx.set_edge_attributes(H, attributesH)
```

Comment:
I believe this can be done without creating any intermediate dict object.
```suggestion
        >>> nx.set_edge_attributes(H, G.edges)
```

Reply:
Sir this actually gives an error. I would need to take it as two values. 

```python
>>> import networkx as nx
>>> G = nx.MultiGraph()
>>> edges = [(0, 1), (0, 1)]
>>> G.add_edges_from(edges)  # Returns list of edge keys
[0, 1]
>>> attributes = {(0, 1, 0): {"cost": 21}, (0, 1, 1): {"cost": 7}}
>>> nx.set_edge_attributes(G, attributes)
>>> H = nx.path_graph([0, 1, 2])
>>> nx.set_edge_attributes(H, G.edges)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/izzza/networkx/networkx/classes/function.py", line 826, in set_edge_attributes
    for (u, v), d in values.items():
ValueError: too many values to unpack (expected 2)

```

Reply:
Ahh... I see....
You are, of course, using a `MultiGraph` for G.
And `H` is not a multigraph. So, then you have to convert the edge from a 3-tuple to a 2-tuple.
Clearly the statement "It doesn't matter the exact type of graph" doesn't quite hold.
For the "usual" case of the same type `nx.set_edge_attributes(H, G.edges)` should just work.
You could change the first line of this example to `H = nx.path_graph([0, 1, 2], create_using=nx.MultiGraph)` unless the point is to show how to use multiedges to update the attributes of edges.

I'd like an example where `nx.set_edge_attributes(H, G.edges)` works.  We can include a second example where conversion from multiedges to edges is shown. But let's start with the more simple case.

The more complex case can be somewhat simplified to:
```python
H = H = nx.path_graph([0, 1, 2])
nx.set_edge_attributes(H, {(u, v): edata for u, v, edata in G.edges(data=True)})
```